core/src/main/java/com/datastrato/gravitino/catalog/CapabilityHelpers.java

```diff
@@ -25,22 +33,148 @@ public static TableChange[] applyCapabilities(Capability capabilities, TableChan
     return Arrays.stream(changes)
         .map(
             change -> {
-              if (change instanceof TableChange.AddColumn) {
-                return applyCapabilities((TableChange.AddColumn) change, capabilities);
+              if (change instanceof TableChange.ColumnChange) {
+                return applyCapabilities((TableChange.ColumnChange) change, capabilities);
 
-              } else if (change instanceof TableChange.UpdateColumnNullability) {
-                return applyCapabilities(
-                    (TableChange.UpdateColumnNullability) change, capabilities);
-
-              } else if (change instanceof TableChange.UpdateColumnDefaultValue) {
-                return applyCapabilities(
-                    ((TableChange.UpdateColumnDefaultValue) change), capabilities);
```

Comment:
Why do you remove this?

Reply:
Do not remove, just move to method `private static TableChange applyCapabilities(
      TableChange.ColumnChange change, Capability capabilities)`: https://github.com/datastrato/gravitino/pull/2975/files#diff-df7bf73c020a2a9709fe067700e70054ab51dd29b071308e03d618b4607fa5a3R158-R168